internal/net/grpc/interceptor/server/logging/accesslog.go

```diff
@@ -0,0 +1,145 @@
+//
+// Copyright (C) 2019-2021 vdaas.org vald team <vald@vdaas.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package logging provides gRPC interceptors for access logging
+package logging
+
+import (
+	"context"
+	"path"
+	"time"
+
+	"github.com/vdaas/vald/internal/log"
+	"github.com/vdaas/vald/internal/observability/trace"
+	"google.golang.org/grpc"
+)
+
+type AccessLogEntity struct {
+	Grpc      *AccessLogGrpcEntity `json:"grpc,omitempty" yaml:"grpc"`
+	StartTime float64              `json:"startTime,omitempty" yaml:"startTime"`
+	Latency   float64              `json:"latency,omitempty" yaml:"latency"`
+	TraceID   string               `json:"traceID,omitempty" yaml:"traceID"`
+	Error     error                `json:"error,omitempty" yaml:"error"`
+}
+
+type AccessLogGrpcEntity struct {
+	Kind    string `json:"kind,omitempty" yaml:"kind"`
+	Service string `json:"service,omitempty" yaml:"service"`
+	Method  string `json:"method,omitempty" yaml:"method"`
+}
+
+func AccessLogInterceptor() grpc.UnaryServerInterceptor {
+	return func(
+		ctx context.Context,
+		req interface{},
+		info *grpc.UnaryServerInfo,
+		handler grpc.UnaryHandler,
+	) (resp interface{}, err error) {
+		var traceID string
+
+		span := trace.FromContext(ctx)
+		if span != nil {
+			traceID = span.SpanContext().TraceID.String()
+		}
+
+		start := time.Now()
+
+		resp, err = handler(ctx, req)
+
+		latency := float64(time.Since(start)) / float64(time.Second)
+		startTime := float64(start.UnixNano()) / float64(time.Second)
+
+		service, method := parseMethod(info.FullMethod)
+
+		entity := &AccessLogEntity{
+			Grpc: &AccessLogGrpcEntity{
+				Kind:    "unary",
+				Service: service,
+				Method:  method,
+			},
+			StartTime: startTime,
+			Latency:   latency,
+		}
+
+		if traceID != "" {
+			entity.TraceID = traceID
+		}
+
+		if err != nil {
+			entity.Error = err
+			log.Error("rpc completed", entity)
```

Comment:
may be you want print entity on error log right?
so, entity print should be %#v print pragma because of entity is pointer type , error log will show thier address only...

Reply:
We should use `log.Error` instead of `log.Errorf` here because Zap logger prints the second argument as the `"details"` field.
https://pkg.go.dev/go.uber.org/zap#Logger.Info
Please see the description of this PR. There's an example.

So you know, there's a known problem (as described in the description) when using Glg logger.

Reply:
Since the log.Error interface is independent of zap, it should not be implemented to work only with zap.
Even if we specify a standard Go logger here, if the group of arguments is the same interface, the address will be displayed as in Glg.
I think it would be more appropriate to consider a more explicit Print method, but I'm not sure.

Reply:
hmm...
I don't know how to print correctly structured JSON logs with that format without using `log.Error`.
Okay, I'll close this PR.

Reply:
wait, you don't need to close this pr.
let's discuss about it.

this pr feature is really effective.